### PR TITLE
[WIP] Alchemy docking deal with edgecases

### PR DESCRIPTION
**Pull request by JenkeScheen**
deals with edge-cases during docking, ultimately letting them fail.

## Developers certificate of origin
- [x] I certify that this contribution is covered by the MIT license as defined in our [LICENSE](https://github.com/choderalab/asapdiscovery/blob/main/LICENSE) and adheres to the [**Developer Certificate of Origin**](https://developercertificate.org/).

### CODE REVIEW

**Comment by JenkeScheen** [comment]
@hmacdope can you have a looksy? this works on my end. Still need to build in a test, but it's possible we'll need to build in a time-out in another PR first because the test on this ligand series will take ~20 min

File: asapdiscovery-alchemy/asapdiscovery/alchemy/cli/prep.py [hunk 1]
```diff
@@ -219,6 +219,7 @@ def alchemize(
     "--dataset-name",
     type=click.STRING,
     help="The name of the AlchemyDataset this will also be the name of the folder created.",
+    required=True,
```
Comment (hmacdope) [inline asapdiscovery-alchemy/asapdiscovery/alchemy/cli/prep.py:222]: Is this needed?

File: asapdiscovery-docking/asapdiscovery/docking/schema/pose_generation.py [hunk 2]
```diff
@@ -692,6 +696,40 @@ def _generate_pose(
...
+
+        Also skips molecules that are larger than `allowed_max_ha` which has been set to a
+        reasonable default. If erroneously large ligands are fed into this docking workflow
+        they will take a prohibitive amount of walltime because of the large number of embeddings
+        to enumerate.
+
+        Returns a success bool, the posed ligand (or input ligand in case of fail) and the
+        error message.
+        """
+        if len(target_ligand.smiles) > allowed_max_ha:
```
Comment (hmacdope) [inline asapdiscovery-docking/asapdiscovery/docking/schema/pose_generation.py:713]: This isn't the number of heavy atoms as includes Hs and stereochem descriptors no?

File: asapdiscovery-docking/asapdiscovery/docking/schema/pose_generation.py [hunk 3]
```diff
@@ -692,6 +696,40 @@ def _generate_pose(
...
+            return (
+                False,
+                target_ligand,
+                f"Query ligand is larger than the allowed number of heavy atoms ({len(target_ligand.smiles)}>{allowed_max_ha}).",
```
Comment (hmacdope) [inline asapdiscovery-docking/asapdiscovery/docking/schema/pose_generation.py:717]: Same here?

**Review by hmacdope — CHANGES_REQUESTED** [review]

File: asapdiscovery-alchemy/asapdiscovery/alchemy/cli/prep.py (hunk 1 above)
Comment (JenkeScheen) [reply asapdiscovery-alchemy/asapdiscovery/alchemy/cli/prep.py:222]: not from this PR, but reminds me that I've wanted to change this for a while. Changing to default "production".

**Comment by JenkeScheen** [comment]
ok added a fast test, should be ready to go once upstream OFFTK issues are fixed @hmacdope

**Comment by hmacdope** [comment]
@JenkeScheen new test is failing. :(

**Review by hmacdope — APPROVED** [review]

**Pull request merged**